security-analysis/security-analysis-api/src/main/java/com/powsybl/security/action/GeneratorAction.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.action;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An action to:
+ * - increase or decrease the targetV of a generator.
+ * - move to voltage control or remove from voltage control.
+ * - change targetQ of a generator that is not a voltage controller.
+ * - change targetV of a generator that is a voltage controller.
+ *
+ * @author Hadrien Godard <hadrien.godard@artelys.com>
+ * @author Anne Tilloy <anne.tilloy@rte-france.com>
+ */
+public class GeneratorAction extends AbstractAction {
+
+    public static final String NAME = "GENERATOR";
+
+    private final String generatorId;
+    private Boolean activePowerRelativeValue; // true if it is a relative variation, false if it is a new targetP
+    private Double activePowerValue; // could be a new targetP if relativeVariation equals false or a relative variation of targetP.
```

Comment:
activePowerRelative**Value** is misleading (could be understood as the relative delta **MW value** if not carefully looking at the type)

```suggestion
    private Boolean activePowerRelative; // true if it is a relative variation, false if it is a new targetP
    private Double activePowerValue; // new targetP if activePowerRelative is false, otherwise a relative change of current targetP value.
```

Reply:
Ok, thanks. I have chosen `activePowerRelativeValue` to remain coherent but already available action (`PhaseTapChangerTapPositionAction`). So what do you think?

Reply:
ok we can leave this as it is